includes/rest-api/class-sensei-rest-api-internal.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Internal REST API for Sensei.
+ *
+ * @package Sensei
+ * @since   3.1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+} // Exit if accessed directly
+
+/**
+ * Class Sensei_REST_API_Internal
+ *
+ * @package rest-api
+ */
+class Sensei_REST_API_Internal {
+
+	/**
+	 * Internal API namespace.
+	 *
+	 * @var string
+	 */
+	private $namespace = 'sensei-internal/v1';
```

Comment:
Besides using the `internal` in the URL, maybe should we also use the `check_ajax_referer`? Or the `permission_callback ` is enough?
If yes and if it takes some time, we could do it in a future PR.

Reply:
It's already part of the Wordpress REST API, without a nonce the user won't be authenticated - `current_user_can` will return false in our case.